R/t-distribution

```diff
@@ -0,0 +1,270 @@
+            style = "position: relative; top: 8px;"
+          )
+        ),
+        div( # calculate p_value
+          class = "d-flex align-items-center",
+          numericInput("p_value", "\\(p\\)-value", STARTING_P_VALUE, 0, 1, 0.01), 
+          actionButton(
+            "p_to_t",
+            "Calculate \\(t\\)-value",
+            class = "btn-primary",
+            style = "position: relative; top: 8px;"
+          )
+        )
+      )
+    )
+  )
+)
+
+# SERVER ---------------------------------------------------------
+server <- function(input, output, session) {
+  crit_lower <- reactive(qt(input$alpha / 2, input$df)) # find lower critical value
+  crit_upper <- reactive(qt(1 - input$alpha / 2, input$df)) # find upper critical value
+  data <- reactive(tibble(
+    t = c(
+      seq(RANGE[1], RANGE[2], PRECISION), 
+      crit_lower() + EXTRA_POINTS_AROUND_CRITICAL_VALUES,
+      crit_upper() + EXTRA_POINTS_AROUND_CRITICAL_VALUES,
+      t_lower() + EXTRA_POINTS_AROUND_CRITICAL_VALUES,
+      t_upper() + EXTRA_POINTS_AROUND_CRITICAL_VALUES
+    ),
+    fill_critical = ifelse(abs(t) >= crit_upper(), "#428BCA99", "transparent"), 
+    fill_p = ifelse(abs(t) >= t_upper(), "#B0306033", "transparent"),
+    d = dt(t, input$df),
+    p = pt(t, input$df),
+  ))
+  
+  mode <- reactive({ # default mode is combined
+    mode <- getQueryString()$mode
+    if (is.null(mode))
+      mode = "combined"
+    mode
+  })
+  output$mode <- reactive(mode())
+  
+  help <- reactive(is.null(getQueryString()$mode))
+  output$help <- reactive(help())
+  output$css <- renderUI({
+    tags$style(HTML("#p-value-panel {", htmltools::css( flex.direction = ifelse( mode() == "combined", "column", "row" ))), "}")
+  })
+  
+  labels <- reactive(tibble(
+    t = c(crit_lower(), crit_upper()),
+    d = dt(t, input$df),
+    label_quantile = glue("t == {round(t, 2)}"),
+    df = input$df,
+    label_percent = label_p_value(t, input$df)
+  ))
+  
+  
+  drawValue <- reactiveVal(FALSE)
+  tValue <- reactiveVal(STARTING_T_VALUE)
+  t_lower <- reactive(-abs(tValue()))
+  t_upper <- reactive(abs(tValue()))
+  pValue <- reactiveVal(STARTING_T_VALUE)
+  
+  ## Calculate p-value based on t: take t_value from input, calculate p-value using pt()
+  onCalculatePValue <- observe({
+    tValue(input$t_value) # take z from input
+    pValue((pt(
+      input$t_value, input$df, lower.tail = input$t_value <= 0
+    ) * 2) %>% round(3))
+    drawValue(TRUE)
+    updateNumericInput(session, "p_value", value = pValue())
+  }) %>% bindEvent(input$t_to_p) 
+  
+  ## Calculate t-value based on p: take p_value from input, calculate t-value using qt()
+  onCalculatetValue <- observe({
+    pValue(input$p_value)
+    tValue(qt(input$p_value / 2, input$df, lower.tail = input$t_value <= 0) %>% round(3))
+    drawValue(TRUE)
+    updateNumericInput(session, "t_value", value = tValue())
+  }) %>% bindEvent(input$p_to_t) 
+  
+  output$critical_values <- renderPrint({
```

Comment:
note that this was unused, but also defective (can't pull 2 variables into a function like that, the function sees it as one variable.)

See update in my patch, and `purrr:pmap`